src/dmd/typesem.d

```diff
@@ -3124,6 +3124,7 @@ Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, int flag)
         {
             if (ident == Id.offsetof)
             {
+                v.dsymbolSemantic(sc);
```

Comment:
The problem here is that `sc` is the context of the expression, which is *not* the same as the context of the field. The solution is to use `null` instead of `sc`.

Reply:
Absolutely, thanks.

Reply:
Done.